prefect_aws/lambda_function.py

```diff
@@ -0,0 +1,191 @@
+            Lambda function will be used.
+        aws_credentials: The AWS credentials to use to connect to AWS Lambda
+            with a default factory of AwsCredentials.
+
+    """
+
+    _block_type_name = "Lambda Function"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
+    _documentation_url = "https://prefecthq.github.io/prefect-aws/s3/#prefect_aws.lambda_function.LambdaFunction"  # noqa
+
+    function_name: str = Field(
+        title="Function Name",
+        description=(
+            "The name, ARN, or partial ARN of the Lambda function to run. This"
+            " must be the name of a function that is already deployed to AWS"
+            " Lambda."
+        ),
+    )
+    qualifier: Optional[str] = Field(
+        default=None,
+        title="Qualifier",
+        description=(
+            "The version or alias of the Lambda function to use when invoked. "
+            "If not specified, the latest (unqualified) version of the Lambda "
+            "function will be used."
+        ),
+    )
+    aws_credentials: AwsCredentials = Field(
+        title="AWS Credentials",
+        default_factory=AwsCredentials,
+        description="The AWS credentials to invoke the Lambda with.",
+    )
+
+    class Config:
+        """Lambda's pydantic configuration."""
+
+        smart_union = True
+
+    def _get_session_and_client(self) -> Tuple[boto3.Session, Any]:
+        """
+        Retrieve a boto3 session and Lambda client
+        """
+        boto_session = self.aws_credentials.get_boto3_session()
+        lambda_client = boto_session.client("lambda")
+        return boto_session, lambda_client
+
+    @sync_compatible
+    async def invoke(
+        self,
+        payload: dict = None,
+        invocation_type: Literal[
+            "RequestResponse", "Event", "DryRun"
+        ] = "RequestResponse",
+        tail: bool = False,
+        client_context: Optional[dict] = None,
+    ) -> dict:
+        """
+        [Invoke](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda/client/invoke.html)
+        the Lambda function with the given payload.
+
+        Args:
+            payload: The payload to send to the Lambda function.
+            invocation_type: The invocation type of the Lambda function. This
+                can be one of "RequestResponse", "Event", or "DryRun". Uses
+                "RequestResponse" by default.
+            tail: If True, the response will include the base64-encoded last 4
+                KB of log data produced by the Lambda function.
+            client_context: The client context to send to the Lambda function.
+                Limited to 3583 bytes.
+
+        Returns:
+            The response from the Lambda function.
+
+        Examples:
+
+            >>> from prefect_aws.lambda_function import LambdaFunction
```

Comment:
nit, but I think we generally do examples like [this](https://github.com/PrefectHQ/prefect-aws/blob/main/prefect_aws/s3.py#L50-L71) instead of the `>>>` syntax. is there specific reason you did that?

Reply:
No particular reason. Will clean it up.